ava-labs/core-mobile
# CP-9566 Notifications doesn't make sound when app in background

## Description

**Ticket: [CP-9566]** 

This fixes issue where notifications won't play custom sound when app is in background.
However, Android requires more development since it requires more created chanels and more logic handling notification permissions.

**Android**
| Event                  | Foreground                         | Background                                                  |
| ------------------------ | ---------------------------------- | ----------------------------------------------------------- |
| BALANCES_SPENT                     | send sound<br>no-notification<br>✅ | receive sound<br>notification<br>✅ ❌ (double notifications) |
| BALANCES_RECEIVED                 | receive sound<br>notification ✅    | receive sound<br>notification ❌ (double notifications)      |
| ALLOWANCE_APPROVED                 | receive sound<br>notification<br>✅ | receive sound<br>notification<br>✅ ❌ (double notifications) |


**iOS**
| Event                      | Foreground                         | Background                         |
| ------------------------ | ---------------------------------- | ---------------------------------- |
| BALANCES_SPENT           | send sound<br>no-notification<br>✅ | send sound<br>notification<br>✅    |
| BALANCES_RECEIVED        | receive sound<br>notification<br>✅ | receive sound<br>notification<br>✅ |
| ALLOWANCE_APPROVED       | default sound<br>notification<br>✅ | default sound<br>notification ✅    |

## Testing
- make sure you can hear notification sound from any other app
- unlock wallet and make sure you are receiving notifications
- put app to background 
- send funds to that wallet
- observe

## Checklist

Please check all that apply (if applicable)


[CP-9566]: https://ava-labs.atlassian.net/browse/CP-9566?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
